Fix remoteManifests

tanner-bruce opened the pull request:
This commit address issue #2160. Remote manifests in the config are
properly pulled again.

This also addresses another unmentioned issue: when using a remote
manifest with `skaffold dev`, the remote manifest will have its image
updated to include the new tag. When shutting down, the remote manifest
will be left the same, including the updated tag. This causes further
runs of `skaffold dev` to no longer match the image, and not update it
properly.

This is addressed by saving all of the images in the manifests found the
first time apply is ran, and then on cleanup replacing the images with
those initially found.

gbird3 commented:
Would love to get this merged in as soon as possible. We want to add skaffold to our development workflow, but due to our current setup this is a requirement before it would make sense to move over.

dgageot commented:
@tanner-bruce any chance you could update this PR?

tanner-bruce commented:
@dgageot @nkubala I've rebased and added a test for cleaning up remote manifests, i.e reverting to original. I feel like there may be something wrong with it but was having some issues figuring out the testing framework. Let me know and I will make changes as necessary. Thanks!

tejal29 commented:
@tanner-bruce i added another test case which you missed.
If the CI is green i will merge this in today.
Thanks for your contribution and we apologize for the delay
